tokio-executor/src/thread_pool/worker.rs

```diff
@@ -270,7 +291,7 @@ where
         }
     }
 
-    fn run_task(&self, task: Task<Shared<P>>) {
+    fn run_task(&mut self, task: Task<Shared<P>>) {
```

Comment:
Somehow the fact that `&mut` is accessed from the thread-local needs to be represented in the API.

Before this change, the unsafety was per file. (going from `&self` -> `&mut self`). Now, the unsafety spans multiple files. Using `&mut self` looks fine in `workers.rs`, but is potentially dangerous due to `current.rs`.

Reply:
I'm not sure I follow? `Entry::owned` still returns `&Owned`, which is the same thing that `current` exposes, so nothing has changed in its interaction with that file?

Reply:
As long as no fn on `Entry` returns `&mut Owned` it should be ok.